nexus-sdk/src/main/java/io/nexusrpc/handler/OperationMiddleware.java

```diff
@@ -0,0 +1,8 @@
+package io.nexusrpc.handler;
+
+/** Middleware for intercepting operations. */
+public interface OperationMiddleware {
+
+  /** Intercepts the given operation. Called once for each operation invocation. */
+  OperationHandler<Object, Object> intercept(OperationHandler<Object, Object> next);
```

Comment:
Yes, but I also wonder if that's enough for future proofing if we wanted to add something else. Maybe a `OperationMiddlewareInterceptContext` that contains the operation context and maybe even this `next` handler (though I'd be ok if they remained separate parameters). Also, I wonder if people should be allowed to alter the input information passed to the next middleware similar to how one might in Temporal interceptors today.

Reply:
Can you address the earlier part of the paragraph about adding some kind of contextual object even if it's empty or mostly empty today? As for altering the values, so long as nobody expects to alter the operation name itself, agreed.

Reply:
>Yes, but I also wonder if that's enough for future proofing if we wanted to add something else.

I think if we want to add anything that is common across all operation methods it would go in the `OperationContext `, if it specific to a certain method like start then the wrapped handler should deal with it.

Reply:
@bergundy if you have any thoughts here

Reply:
I agree we can supply some context here. It would contain the service name, operation name, and request header. The rest of the information is method specific, and IMHO, should be intercepted by the wrapped handler.

Reply:
The obvious downside for instantiating these handlers per request is the extra garbage generated but I'm not too concerned with that.